### DISCUSSION TITLE
Compiling into mainline source not clear

### DISCUSSION THREAD

**Discussion by roadapathy** [NVIDIA/open-gpu-kernel-modules#343] [General]
Sorry, the how to does not explain how to integrate this into the Kernel. Does the code directory not go somewhere into the Kernel source directory?  I compile and install the Kernel all the time from kernel.org but this isn't clear to me.

**Comment by YusufKhan-gamedev** [comment]
Use `modprobe`, compile this with whats said in the README then use modprobe, make sure to blacklist nouveau beforehand.

**Comment by mtijanic** [comment]
Hello! The codebase is for an out-of-tree driver that is compiled separately from the mainline kernel. Simply running `make modules` from the top level directory of this repo should build all the necessary `.ko` files.

You can then install these with `make modules_install` which will copy them into the right place and you can load with `modprobe` or just by running e.g. `nvidia-smi`.

This codebase is generally not suitable for integration into the mainline kernel. You probably can hack it into the build system locally (e.g. copy into `drivers/gpu/drm/nvidia`, hook into Kbuild), but that is something we have never tested.
